src/cript/api/api.py

```diff
@@ -0,0 +1,234 @@
+import copy
+import os
+import warnings
+from typing import List, Literal, Union
+
+from cript.api._valid_search_modes import _VALID_SEARCH_MODES
+from cript.api.exceptions import CRIPTAPIAccessError, CRIPTConnectionError
+from cript.nodes.primary_nodes.primary_base_node import PrimaryBaseNode
+from cript.nodes.primary_nodes.project import Project
+from cript.nodes.supporting_nodes.group import Group
+from cript.nodes.supporting_nodes.user import User
+
+_global_cached_api = None
+
+
+def _get_global_cached_api():
+    """
+    Read-Only access to the globally cached API object.
+    Raises an exception if no global API object is cached yet.
+    """
+    if _global_cached_api is None:
+        raise CRIPTAPIAccessError
+    return copy.copy(_global_cached_api)
```

Comment:
Python doesn't really have the concept of constant objects.
Not like C++ for example.
So, I can't really make it constant.

The best approximation we have is to return a copy, since there is never write access to the globally cached instance.

Reply:
I looked into making the function static a while ago and from what I remembered it didn't make sense to have the host be dynamic, but the `_get_host()` be static. However, if I am wrong this could be a good way to go about it, by just making a single method static that always gets the host and the `validate_schema` and `validate_vocab` can just get the host and make their own calls to the API to get the needed resources and return either a True or return an Exception.

Reply:
I not of fan of using copy here, I would just return the object itself. If a users messing with a private object (object with leading underscore) they do so knowing the will possibly break the code (this is Python general practice).  
If you do want to make it constant; it is typically done by making it a frozen class.